hbase/src/main/java/org/apache/metamodel/hbase/HBaseClient.java

```diff
@@ -0,0 +1,207 @@
+
+/**
+ * This class can perform client-operations on a HBase datastore
+ */
+final class HBaseClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(HBaseClient.class);
+
+    private final Connection _connection;
+
+    public HBaseClient(final Connection connection) {
+        this._connection = connection;
+    }
+
+    /**
+     * Insert a single row of values to a HBase table.
+     * @param tableName
+     * @param columns
+     * @param values
+     * @throws IllegalArgumentException when any parameter is null or the indexOfIdColumn is impossible
+     * @throws MetaModelException when no ID-column is found.
+     * @throws MetaModelException when a {@link IOException} is caught
+     */
+    public void insertRow(final String tableName, final HBaseColumn[] columns, final Object[] values,
+            final int indexOfIdColumn) {
+        if (tableName == null || columns == null || values == null || indexOfIdColumn >= values.length
+                || values[indexOfIdColumn] == null) {
+            throw new IllegalArgumentException(
+                    "Can't insert a row without having (correct) tableName, columns, values or indexOfIdColumn");
+        }
+        if (columns.length != values.length) {
+            throw new IllegalArgumentException("The amount of columns don't match the amount of values");
+        }
+        try (final Table table = _connection.getTable(TableName.valueOf(tableName))) {
+            // Create a put with the values of indexOfIdColumn as rowkey
+            final Put put = new Put(getValueAsByteArray(values[indexOfIdColumn]));
+
+            // Add the other values to the put
+            for (int i = 0; i < columns.length; i++) {
+                if (i != indexOfIdColumn) {
+                    // NullChecker is already forced within the HBaseColumn class
+                    final byte[] columnFamily = Bytes.toBytes(columns[i].getColumnFamily());
+                    // An HBaseColumn doesn't need a qualifier, this only works when the qualifier is empty (not
+                    // null). Otherwise NullPointer exceptions will happen
+                    byte[] qualifier = null;
+                    if (columns[i].getQualifier() != null) {
+                        qualifier = Bytes.toBytes(columns[i].getQualifier());
+                    } else {
+                        qualifier = Bytes.toBytes(new String(""));
+                    }
+                    final byte[] value = getValueAsByteArray(values[i]);
+                    // A NULL value, doesn't get inserted in HBase
+                    // TODO: Do we delete the cell (and therefore the qualifier) if the table get's updated?
+                    if (value == null) {
+                        logger.warn("The value of column '{}:{}' is null. This insertion is skipped", columns[i]
+                                .getColumnFamily()
+                                .toString(), columns[i].getQualifier().toString());
```

Comment:
It is possible that an HBaseColumn doesn't have a qualifier, in which case this will throw a NullPointerException, so either do `Bytes.toString(qualifier)`, or an inline if to check if the qualifier is null and in that case return "" for it.